web/js/palettes/legend.js

```diff
@@ -116,10 +117,10 @@ export function palettesLegend(spec) {
   };
 
   var updateClasses = function (legend, index) {
+    var $colorBox;
```

Comment:
You can add the `var` keyword two lines down where `$colorBox` is actually defined.

Reply:
There are two each methods containing `$colorbox`, so this var needs to be defined at the top in this function.

Reply:
It looks like those are two different `$colorBox` variables that just happen to have the same name. Are they supposed to refer to the exact same element? Doesn't making them use the same variable increase the risk that one might overwrite the other?

Reply:
That existing lodash each loop is a little strange. It looks like the same loop happens twice;
```
loEach(legend.colors, function (color, classIndex) {
```
That could probably be simplified to only happen once.

Reply:
Yea, this is weird. It seems like these two can be combined, I'll look at combining these.